src/scanner.zig

```diff
@@ -144,12 +146,38 @@ const Protocol = struct {
 
     fn parse(allocator: *mem.Allocator, parser: *xml.Parser) !Protocol {
         var name: ?[]const u8 = null;
+        var copyright: ?[]const u8 = null;
+        var toplevel_description: ?[]const u8 = null;
         var interfaces = std.ArrayList(Interface).init(allocator);
         while (parser.next()) |ev| switch (ev) {
             .open_tag => |tag| {
-                // TODO: parse copyright and description
-                if (mem.eql(u8, tag, "interface"))
+                if (mem.eql(u8, tag, "copyright")) {
+                    if (copyright != null)
+                        return error.DuplicateCopyright;
+                    if (parser.next()) |e| {
+                        if (e == .character_data) {
+                            copyright = try allocator.dupe(u8, e.character_data);
+                        } else {
+                            return error.BadCopyright;
+                        }
+                    } else {
+                        return error.UnexpectedEndOfFile;
+                    }
+                } else if (mem.eql(u8, tag, "description")) {
+                    if (toplevel_description != null)
+                        return error.DuplicateToplevelDescription;
```

Comment:
Not sure if this is right. Are multiple toplevel descriptions allowed? I have never encountered it so far and it does not make much sense anyway.

Reply:
Nope, only one is allowed and it must come after the copyright and before any interfaces: https://gitlab.freedesktop.org/wayland/wayland/-/blob/master/protocol/wayland.dtd